ouroboros-consensus-cardano/tools/db-analyser/Main.hs

```diff
@@ -0,0 +1,327 @@
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+
+module Main (main) where
+
+import           Control.Monad.Except
+import qualified Data.ByteString as BS
+import           Data.Either (fromRight)
+import           Data.Foldable (asum)
+import           Data.IORef
+import           Data.List (intercalate)
+import           Data.Proxy (Proxy (..))
+import qualified Data.Text as Text
+import           GHC.Natural (Natural)
+import           Options.Applicative
+import           System.FilePath.Posix ((</>))
```

Comment:
```suggestion
import           System.FilePath ((</>))
```

You should import the Posix one directly, you lose cross-OS compatibility. Import `System.FilePath` and either `.Posix` or `.Windows` will be used depending on the OS.

(Sorry, I linked to the wrong one in my comment, that's because it turns up first on Hoogle.)